tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc

```diff
@@ -65,10 +116,14 @@ __global__ void SortedSegmentSumCustomKernel(
     T sum = T(0);
     Index first_segment_id = segment_ids[input_outer_dim_index_base];
     Index last_output_segment_id = output_outer_dim_size;
+    bool is_first_update_racy = input_outer_dim_index_base > 0
+        && first_segment_id == segment_ids[input_outer_dim_index_base - 1]
```

Comment:
What if `input_outer_dim_index_base` is `0`?  If it can't be `0` then please add a comment stating why.